### ISSUE TITLE
Cashmere Release Date

### ISSUE THREAD

**Issue by jtrujill** [HealthCatalyst/Fabric.Cashmere#231]
I’ve heard from people that Cashmere was supposed to be released either last Friday or this coming Friday and I just want to put it out there that I don’t think Cashmere is close enough to be at a 1.0 release. There are still some serious issues being worked out, and I think it’d be best to put a hold on the release for a bit.

I'd like to get all components reviewed and really go over the interfaces, because once something is released, it'll be difficult to remove stuff from those interfaces. For example, I noticed the input element doesn't make use of the classes(ng-invalid, ng-pristine,ng-touched,ng-dirty) that are added to it when you use Angular forms. Instead there's a highlight input which is used to do something similar.

I don't think there should be a huge rush to push something out that isn't ready, and I'd like to delay the release until around mid to late May. If we release a buggy and unstable interface for developers to use and then we sweep the interface from under their feet, there will be less willingness to use this library. 

We could also use this time to get some tests written for the library. I'll be going over all of the components that I wrote to add in tests now that the ACO team is putting more work into this product

**Comment by andrew-frueh** [comment]
I agree with you @jtrujill - I've been thinking the same thing.  A lot has changed for Cashmere in the past 10 days with your team starting to use it in your app.  In many ways the 1.0 release was more about a list of components, but as we've discovered, a big component list does not a successful library make.  At best we're really only at a beta phase right now.  The formation of this steering committee is a huge step forward that wouldn't have come about if it wasn't for you guys putting it through its paces and pointing out weak spots.  

Bottom line, I'm in agreement with you that we don't have to race to get the library to 1.0.  Let's keep doing what we're doing for now, and in the committee's first meeting a couple weeks from now, that group can agree on a more formal product/project plan and decide what is required for a true 1.0.

**Comment by TonyCurtisLives** [comment]
So we were wanting to implement the library in early May in edw ops. Does this mean the library is too "buggy and unstable" for us to go forward with this? What serious issues are there? It is something I would like to know before we plan to move forward.

**Comment by jtrujill** [comment]
@TonyCurtisLives The interface of the components is likely to change, so if you use the library don't be too surprised if things break on you. Things are still being worked out. I'm going to be updating the input component which will introduce a slight breaking change. 

The library doesn't have many unit tests and for me that is a concern and will be the next thing I will be tackling. I just don't want you to begin using the library and get frustrated that if it creates extra work for you and your team.

I would say it's in an alpha phase close towards beta, so I'd use it sparingly or just hold of for a little bit. I'm not sure where the library will be by mid May, but I'm thinking the earliest we could have a 1.0.0 release would be end of May.

**Comment by andrew-frueh** [comment]
Good question @TonyCurtisLives and thanks @jtrujill for the follow up.  From a product management position, if your team is willing to be early adopters, Tony, it will be great for Cashmere - even if it means potentially a bit more work for your team to log issues when they come up, and potentially help submit PR's for fixes.  We've learned a lot from having the CQE folks get in there and start using it, and having another team do the same would be equally helpful.

Also on the flip side, if you *don't* use Cashmere, you're going to end up writing your own version of these components (which is not a trivial amount of time), and you'll potentially need to switch yours over to the Cashmere versions down the line.  So the additional time you have to spend as an early adopter might come out in the wash.

**Comment by TonyCurtisLives** [comment]
So what I hear you saying @andrew-frueh is the team effort to early adopt will not likely cost us more time than build from scratch. But I can see the value in the effort being more positive in that the effort will benefit many rather than a single project. To me this sounds reasonable, and I do want to support Cashmere, so we will go ahead and use the libraries.

**Comment by andrew-frueh** [comment]
The Steering Committee had a chance to discuss this on Wednesday and agreed that Cashmere is currently in an alpha release state.  I've [captured the current thinking on the wiki](https://github.com/HealthCatalyst/Fabric.Cashmere/wiki/Steering-Committee-Decisions#current-release), so I'm going to close this issue and we'll update the release schedule as soon as we know more.
